src/memote/reactions.jl

```diff
@@ -4,3 +4,115 @@ M.E., Olivier, B.G. et al. MEMOTE for standardized genome-scale metabolic model
+    end
+    return metabolic_reactions
+end
+
+"""
+$(TYPEDSIGNATURES)
+
+Find all transport reactions and indicate which are constrained beyond
+directionality. For a constraint to be not purely directional, the lower or
+upper bound needs to be different from `[-config.reactions.bound_default, 0,
+config.reactions.bound_default]`. Transport reactions are heuristically
+identified, see [`!_probably_transport_reaction`](@ref), which partially uses
+reaction annotations. Set the annotation field to use via
+`config.reactions.rest_annotation`. 
+"""
+function find_all_transport_reactions(model; config = memote_config)
+    transport_reactions = Dict{String, Bool}()
+    for (lb, ub, rid) in zip(bounds(model)..., reactions(model))
+        !_probably_transport_reaction(model, rid, config.reaction.test_annotation) && continue
+        transport_reactions[rid] = _is_constrained(lb, ub, config.reaction.bound_default)
+    end
+    return transport_reactions
+end
+
+"""
+$(TYPEDSIGNATURES)
+
+Find all reactions with overlapping annotation information. Internally calls
+[`annotation_index`](@ref). Some annotations, like sbo terms will necessarily be
+non-unique, ignore annotations like this by editing `config.reaction.ignore_annotations`.
+"""
+function reactions_with_partially_identical_annotations(model; config = memote_config)
+    stdmodel = convert(StandardModel, model)
+    idx = annotation_index(stdmodel.reactions)
+    for anno in config.reaction.ignore_annotations
+        delete!(idx, anno)
+    end
+    ambiguously_identified_items(idx)
+end
+
+"""
+$(TYPEDSIGNATURES)
+
+Return a list of all reactions that are duplicated.
+"""
+function duplicate_reactions(model)
+    stdmodel = convert(StandardModel, model)
+    duplicated_rxns = Set{String}()
+    for rid in reactions(stdmodel) 
+        dup = check_duplicate_reaction(
+            stdmodel.reactions[rid],
+            stdmodel.reactions;
+            only_metabolites = false,
+        )
+        isnothing(dup) && continue
+        push!(duplicated_rxns, dup)
+    end
+    return duplicated_rxns
+end
+
+"""
+$(TYPEDSIGNATURES)
+
+Identify reactions that have the same gene reaction rules. Does not take
+directionality or compartments into account.
+"""
+function reactions_with_identical_genes(model)
+    grr_rids = Dict{Vector{String}, Set{String}}()
+    rids = reactions(model)
+    for rid1 in rids
+        !_has_sensible_gpr(model, rid1) && continue
+        grrs1 = reaction_gene_association(model, rid1)
+        for rid2 in rids
+            rid1 == rid2 && continue
+            !_has_sensible_gpr(model, rid2) && continue
+            grrs2 = reaction_gene_association(model, rid2)
+            
+            for grr1 in grrs1
+                for grr2 in grrs2
+                    issetequal(grr1, grr2) && push!(get!(grr_rids, sort(grr1), Set{Vector{String}}()), rid1)
+                end
+            end
+        end 
+    end
+    return grr_rids
```

Comment:
this can be done more transparently
1] blindly push every reaction to a dictionary keyed by the gene sets
2] filter it so that only gene sets with ≥ reactions remain

Reply:
ah yes, this is much better